wiki/Data-Access-Modeling.md

```diff
@@ -0,0 +1,188 @@
+}
+```
+
+```cs
+public class InfractionSummary
+{
+    ...
+
+    [ExpansionExpression]
+    internal static readonly Expression<Func<InfractionEntity, InfractionSummary>> FromEntityProjection
+        = entity => new InfractionSummary()
+        {
+            Id = entity.Id,
+            GuildId = entity.GuildId,
+            Type = entity.Type,
+            Reason = entity.Reason,
+            Duration = entity.Duration,
+            Subject = entity.Subject.Project(GuildUserBrief.FromEntityProjection),
+            CreateAction = entity.CreateAction.Project(ModerationActionBrief.FromEntityProjection),
+            RescindAction = (entity.RescindAction == null)
+                ? null
+                : entity.RescindAction.Project(ModerationActionBrief.FromEntityProjection),
+            DeleteAction = (entity.DeleteAction == null)
+                ? null
+                : entity.DeleteAction.Project(ModerationActionBrief.FromEntityProjection),
+            Expires = entity.CreateAction.Created + entity.Duration
+        };
+}
+```
+
+```cs
+internal static class InfractionQueryableExtensions
+{
+    public static IQueryable<InfractionEntity> FilterBy(this IQueryable<InfractionEntity> query, InfractionSearchCriteria criteria)
+        => query
+            .FilterBy(
+                predicate:  x => x.Id == criteria.Id,
+                criteria:   criteria?.Id != null)
+            .FilterBy(
+                predicate:  x => x.GuildId == criteria!.GuildId,
+                criteria:   criteria?.GuildId != null)
+            .FilterBy(
+                predicate:  x => criteria!.Types.Contains(x.Type),
+                criteria:   criteria?.Types?.Any() ?? false)
+            ...
+}
+```
+
+### Transactions
+
+I'm not going to bother actually describing the `IRepositoryTransaction` system, because it's trash. It was built to serve a synchronization need, and does accomplish that, but it's unintuitive and clunky compared to `System.Transactions`, which is what we need to move to.
+
+Suffice it to say that operations within the Business Logic or Data Access layers should be wrapped within an appropriate transaction whenever they A) perform multiple Data Access operations that depend upon one another (like the insertion of parent and child records), or B) perform READ operations whose results affect subsequent WRITE operations.
+
+### Testing
+
+MODiX utilizes the EF Core InMemory database provider to perform testing. This provider simulates the presence of a real underlying database provider, by maintaining entities within in-memory collections. This provider does have several drawbacks, though. It does not support raw SQL queries at all, and is not simulate a "Relational" database. This last bit is important, because it means that constraints enforced by a real RDBMS, such as Foriegn Key constraints, are not enforced by the InMemory provider. Thus there are categories of bugs that cannot be caught through use of the InMemory provider.
```

Comment:
Typo: "and is not simulate" -> "and does not simulate"

```suggestion
MODiX utilizes the EF Core InMemory database provider to perform testing. This provider simulates the presence of a real underlying database provider, by maintaining entities within in-memory collections. This provider does have several drawbacks, though. It does not support raw SQL queries at all, and does not simulate a "Relational" database. This last bit is important, because it means that constraints enforced by a real RDBMS, such as Foriegn Key constraints, are not enforced by the InMemory provider. Thus there are categories of bugs that cannot be caught through use of the InMemory provider.
```